OpenRA.Mods.Common/Traits/Explodes.cs

```diff
@@ -54,9 +55,13 @@ public void Killed(Actor self, AttackInfo e)
 				return;
 
 			var weaponName = ChooseWeaponForExplosion(self);
-			if (weaponName == null)
+			if (string.IsNullOrEmpty(weaponName))
 				return;
 
+			if (!e.Attacker.World.Map.Rules.Weapons.ContainsKey(weaponName.ToLowerInvariant()))
```

Comment:
The `WeaponReference` already gives us this, so i'm not convinced we need this.  I guess this is going against my comments in #8940 however, so feel free to ignore me.